prometheus/prometheus.libsonnet

```diff
@@ -38,17 +39,26 @@ local kausal = import 'ksonnet-util/kausal.libsonnet';
       + configMap.withData({
         'prometheus.yml': k.util.manifestYaml(prometheus_config),
       }),
-
-      configMap.new('%s-alerts' % _config.name)
-      + configMap.withData({
-        'alerts.rules': k.util.manifestYaml(prometheusAlerts),
-      }),
-
-      configMap.new('%s-recording' % _config.name)
-      + configMap.withData({
-        'recording.rules': k.util.manifestYaml(prometheusRules),
-      }),
-    ],
+    ]
+    + (
+      if std.objectHas(this, 'prometheusAlerts') && std.prune(this.prometheusAlerts) != {}
```

Comment:
nit: this code repeats few times, it _might_ live better alone in a separate function (or not, no strong opinion here).

Reply:
Agreed, I want to consolidate this further in the near future.